src/components/ListTodos/ListTodos.js

```diff
@@ -0,0 +1,148 @@
+import React from 'react';
+import PropTypes, { object } from 'prop-types';
+
+class ListTodos extends React.Component {
+  state = {
+    editingId: '',
+    editValue: '',
+    errorAddNewValue: false,
+  }
+
+  editingValue = (id, value) => {
+    this.setState(() => ({
+      editingId: id, editValue: value,
+    }));
+  }
+
+  changeTodoValue = (e) => {
+    const letter = e.target.value.replace(/^ /, '');
+
+    this.setState({ editValue: letter });
+  }
+
+  lostFocus = () => {
+    const { changeTodoValue } = this.props;
+
+    this.setState((state) => {
+      const editingId = '';
+      const editValue = '';
+      let errorAddNewValue = false;
+
+      if (state.editValue.length < 3) {
+        errorAddNewValue = true;
+
+        return {
+          errorAddNewValue,
+        };
+      }
+
+      changeTodoValue(state.editingId, state.editValue);
+
+      return {
+        errorAddNewValue,
+        editingId,
+        editValue,
+      };
+    });
+  }
+
+  saveOrCancel = (e) => {
+    const { changeTodoValue } = this.props;
+    const currentKey = e.key;
+
+    this.setState((state) => {
+      const errorAddNewValue = false;
+      const editingId = '';
+      const editValue = '';
+
+      if (currentKey === 'Enter' && state.editValue.length >= 3) {
+        changeTodoValue(state.editingId, state.editValue);
+
+        return {
+          editingId, editValue, errorAddNewValue: false,
```

Comment:
it's not convenient when you write all values in one line, move each of them on the new line